File: tests/ai-fixtures/vibecheck-ai.rb
# ruleid: vibecheck-untrusted-input-in-system-prompt-rb
msg = { role: "system", content: "You are #{params[:persona]}" }
def ex(client)
  resp = client.messages.create(model: "m")
  # ruleid: vibecheck-llm-output-executed-rb
  system(resp.content)
end
def html(client)
  r = client.messages.create(model: "m")
  # ruleid: vibecheck-llm-output-raw-html-rb
  raw(r.content)
end

File: tests/ai-fixtures/vibecheck-ai.py
from anthropic import Anthropic
from flask import request
import subprocess
client = Anthropic()
def persona():
    # ruleid: vibecheck-untrusted-input-in-system-prompt-py
    client.messages.create(model="m", system=f"You are {request.json['persona']}", messages=[])
def ex():
    resp = client.messages.create(model="m", messages=[])
    # ruleid: vibecheck-llm-output-executed-py
    subprocess.run(resp.content[0].text)
def html():
    r = client.messages.create(model="m", messages=[])
    # ruleid: vibecheck-llm-output-raw-html-py
    return HTMLResponse(r.content[0].text)
